api/rest/client.go

```diff
@@ -96,10 +96,14 @@ func (c *Client) DoRequest(req *http.Request, resp interface{}) (statusCode int,
 		httpError := struct {
 			Message string `json:"message"`
 		}{}
-		err = json.NewDecoder(httpResp.Body).Decode(&httpError)
+		body, err := io.ReadAll(httpResp.Body)
 		if err != nil {
 			return httpResp.StatusCode, err
 		}
```

Comment:
Should we be returning an error from reading the response body along with the status code? It seems a little strange.

For example, we could have the server respond with a 200 OK, but for some reason, fail to read the body of the response. We'd then return `200` with an error message, unrelated to the HTTP protocol.

The way I see it, we should either return:
* 2xx HTTP status code - no error
* 4xx-5xx HTTP status code - HTTP error (from body or from status text)
* HTTP status code zero value - non-HTTP error